src/qmxgraph/page/utils.js

```diff
@@ -66,16 +66,58 @@ graphs.utils.setStyleKey = function setStyleKey (style, key, value) {
         }
         else
         {
-            var cont = style.indexOf(';', index);
+            var parts = style.split(';')
+            for (var i = 0; i < parts.length; ++i)
+            {
+                var current = parts[i];
+                var current_key = current.split('=')[0];
+                if (current_key == key)
+                {
+                    break; // Just to find the key index
+                }
+            }
 
-            style = style.substring(0, index) + key + '=' + value +
-                ((cont >= 0) ? style.substring(cont) : '');
+            var updated_key_value = key + '=' + value;
+            parts.splice(i, 1, updated_key_value)
+            style = parts.join(';');
         }
     }
 
     return style;
 };
 
+/**
+ * Removes the given key in an inline style.
+ *
+ * @param {string} style An inline style of mxGraph.
+ * @param {string} key Key in style.
+ * @returns {string} New inline style with the key removed.
+ */
+graphs.utils.removeStyleKey = function removeStyleKey (style, key) {
```

Comment:
We should unit test this (possibly calling `eval_js` from Python). Same for `setStyleKey`.